cmds/tcz/tcz.go

```diff
@@ -87,19 +88,26 @@ func clonetree(tree string) error {
 			return nil
 		}
 		// all else gets a symlink.
-		if link, err := os.Readlink(path); err == nil {
+
+		// If the link exists
+		if link, err := os.Readlink(path[lt:]); err == nil {
+			// Confirm that it points to the same path to be assigned
 			if link == path {
 				return nil
 			}
+
+			// If it does not, return error because tcz packages are inconsistent
 			l.Printf("Symlink: need %v -> %v but %v -> %v is already there", path, path[lt:], path, link)
 			return err
 		}
-		l.Printf("Need to symlnk %v to %v\n", path, path[lt:])
+		// If the link does not exist
+		l.Printf("Need to symlink %v to %v\n", path, path[lt:])
+
 		if err := os.Symlink(path, path[lt:]); err != nil {
-			// TODO: if it's there, and has same value, no error.
-			l.Printf("symlink failed: %v", err)
+			l.Printf("Symlink: %v", err)
```

Comment:
It's customary in Go to wrap errors instead of printing a warning and returning the original error.

```go
if err := os.Symlink(path, path[lt:]); err != nil {
  return fmt.Errorf("symlink(%q ->  %q) = %v", path[lt:], path, err)
}
return nil
```

The caller may then decide what to do with the error - whether that's printing it or discarding it or passing it on is up to them.